test/smtp.test.lua

```diff
@@ -143,9 +165,50 @@ test:test("smtp.client", function(test)
 
     m = mails:get()
     subj = select(2, string.gsub(
-                  m.text,
-                  "Subject: =%?utf%-8%?b%?YWJjZGVmZ2hpamvRj2xtbm9wcXJzdHV2d3h5eg==%?=", ""))
+                        m.text,
+                "Subject: =%?utf%-8%?b%?YWJjZGVmZ2hpamvRj2xtbm9wcXJzdHV2d3h5eg==%?=", ""))
     test:is(subj, 1, 'subject codes >127')
+
+    r = client:request(addr, '3xx@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'MAIL failed: 354 (SMTP error)', 'Errors 3xx')
+
+    r = client:request(addr, '4xx@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'MAIL failed: 451 (SMTP error)', 'service unavailable')
+
+    r = client:request(addr, '5xx@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'MAIL failed: 510 (SMTP error)', 'unexisting recipient')
+
+    r = client:request(addr, '10xx@tarantool.org',
+                       'receiver@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'response reading failed (SMTP error)', 'unexisting recipient')
+
+    r = client:request(addr, 'sender@tarantool.org',
+                       '3xx@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'RCPT failed: 354 (SMTP error)', 'Errors 3xx')
+
+    r = client:request(addr, 'sender@tarantool.org',
+                       '4xx@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'RCPT failed: 421 (SMTP error)', 'service unavailable')
+
+    r = client:request(addr, 'sender@tarantool.org',
+                       '5xx@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'RCPT failed: 510 (SMTP error)', 'unexisting recipient')
+
+    r = client:request(addr, 'sender@tarantool.org',
+                       '10xx@tarantool.org',
+                       'mail.body')
+    test:is(r.reason, 'response reading failed (SMTP error)', 'unexisting recipient')
```

Comment:
That looks more like receiving malformed answer, cause, for example, `1000` status code is not typicall for SMTP that is why there are some troubles with handling this type of repsonses. For now this test is about another failre, so that description is changed, cause there is no handling of malformed answers yet. I'm working on it currently.